### PR TITLE
Catch parse json object exception using GSON

### PR DESCRIPTION
**Pull request by wangdaliu**

### CODE REVIEW

File: affirm/src/main/java/com/affirm/android/AffirmTracker.java [hunk 1]
```diff
@@ -66,13 +68,16 @@ static void track(@NonNull TrackingEvent event, @NonNull TrackingLevel level,
...
-
-        fillTrackingData(eventName, data, level);
-        return data;
+        try {
+            final JsonObject data = eventData == null ? new JsonObject()
+                    : gson.fromJson(gson.toJson(eventData, JsonObject.class), JsonObject.class);
+
+            fillTrackingData(eventName, data, level);
+            return data;
+        } catch (JsonSyntaxException e) {
```
Comment (kevindavmoore) [inline affirm/src/main/java/com/affirm/android/AffirmTracker.java:78]: Should this also have the JsonIOException?

File: affirm/src/main/java/com/affirm/android/TrackerRequest.java [hunk 2]
```diff
@@ -56,17 +59,20 @@ public void onResponse(@NotNull Call call, @NotNull Response response) {
...
+                            final AffirmError affirmError = AffirmPlugins.get()
+                                    .gson()
+                                    .fromJson(responseBody.charStream(), AffirmError.class);
+                            AffirmException affirmException = AffirmHttpClient.handleAPIError(
+                                    affirmError,
+                                    response.code(),
+                                    response.headers().get(X_AFFIRM_REQUEST_ID)
+                            );
+                            handleException(affirmException);
+                        } catch (JsonSyntaxException | JsonIOException e) {
```
Comment (kevindavmoore) [inline affirm/src/main/java/com/affirm/android/TrackerRequest.java:72]: Should this be an IOException?

**Comment by shangeethsivan** [comment]
@wangdaliu @vcello can I know in which release we can see this change.. We are seeing a spike in the crash 
https://github.com/Affirm/affirm-merchant-sdk-android/issues/71

**Comment by vcello** [comment]
@shangeethsivan this fix will be in the next release, targeted for this week.

**Pull request merged**